### PR TITLE
fix(ActiveSelection): make sure canvas is in charge of setting initial coords

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Moved the call to `setCoords` from the constructor.
It should not be there because the canvas ref is being set after that possibly, leading to wrong coords due to vpt.

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
ready for review

**Review by ShaMan123 — COMMENTED** [review]
this fix is ready
the test is broken on master

**Comment by asturur** [comment]
The changes is ok for me, but we should comunicate somewhere that using ActiveSelection manually is becoming highly not suggested.

File: src/shapes/ActiveSelection.spec.ts [hunk 1]
```diff
@@ -55,4 +56,20 @@ describe('ActiveSelection', () => {
...
+        new FabricObject({
+          left: 100,
+          top: 100,
+          width: 100,
+          height: 100,
+        }),
+      ]),
+      viewportTransform: [2, 0, 0, 0.5, 400, 150],
+    });
+    expect(canvas.getActiveSelection().getCoords()).toMatchSnapshot();
```
Comment (asturur) [inline src/shapes/ActiveSelection.spec.ts:72]: let's test by picking aCoords and lineCoords that is what are we testing today.
We are testing that those are correct, without the need of being recalculated.
getCoords return an array and is mostly for internal usage of calculations, if we snapshot aCoords and lineCoords we also see tl,tr,bl,br.

File: src/shapes/ActiveSelection.spec.ts [hunk 2]
```diff
@@ -55,4 +56,20 @@ describe('ActiveSelection', () => {
     selection.add(new FabricObject({ left: 50, top: 50, strokeWidth: 0 }));
     expect(selection.item(0).getCenterPoint()).toEqual({ x: 50, y: 50 });
   });
+
+  it('sets coords after attaching to canvas', () => {
```
Comment (asturur) [inline src/shapes/ActiveSelection.spec.ts:60]: Isn't this a canvas test? is the canvas executing the new code no?

Comment (ShaMan123) [reply src/shapes/ActiveSelection.spec.ts:60]: true, I have no idea where it should belong

File: src/shapes/ActiveSelection.spec.ts (hunk 1 above)
Comment (ShaMan123) [reply src/shapes/ActiveSelection.spec.ts:72]: > We are testing that those are correct, without the need of being recalculated.

They are not recalcuated.

I just have in mind #8767 so I avoid lineCoords already but whatever

**Comment by ShaMan123** [comment]
> The changes is ok for me, but we should comunicate somewhere that using ActiveSelection manually is becoming highly not suggested.

I was thinking to make the constructor protected. Then you get a TS error you can ignore.

**Review by ShaMan123 — COMMENTED** [review]
ok

File: src/shapes/ActiveSelection.spec.ts (hunk 1 above)
Comment (asturur) [reply src/shapes/ActiveSelection.spec.ts]: they are not recalculated no, but no owe knows it, is not written the test is not explcit.
if getCoords change or recalculation becomes default, this test doesn't break but also becomes uneffective.

**Review by asturur — APPROVED** [review]

**Review by asturur — APPROVED** [review]

File: src/shapes/ActiveSelection.spec.ts (hunk 1 above)
Comment (ShaMan123) [reply src/shapes/ActiveSelection.spec.ts]: I do not follow
What should be changed?

Comment (asturur) [reply src/shapes/ActiveSelection.spec.ts]: oh i thought you changed because i have seen outdated tag on the discussion
i ll change it never mind

**Pull request merged**